store.go

```diff
@@ -1,51 +1,70 @@
 package vstore
 
-// Store is the main redux-like store. Use it to access the State properties
-// through the ComponentConstructor.
-type Store struct {
-	r    Reducer
-	mws  []Middleware
-	subs map[*SubFunc]bool
+import (
+	"github.com/gopherjs/vecty"
+	"sync"
+)
+
+// store is the main redux-like store. Use it to access the State properties.
+type Store interface {
+	Connect(comp StoreComponent) StoreComponent
+	Dispatch(action interface{})
+	State() interface{}
 }
 
-// SubFunc is a subscription callback.
-type SubFunc func(s *Store)
+// store satisfies Store.
+type store struct {
+	r   Reducer
+	mws []Middleware
+
+	mtx   sync.RWMutex
+	comps map[*storeComponent]bool
+}
 
 // Middleware is a simple middleware that gets called before any Reduce calls.
 type Middleware func(action interface{})
 
-// CreateStore returns a store that you can make global dispatches to in order
+// New returns a store that you can make global dispatches to in order
 // to update the passed global state.
-func CreateStore(r Reducer, mws ...Middleware) *Store {
-	s := Store{r: r, mws: mws, subs: map[*SubFunc]bool{}}
-
-	return &s
+func New(r Reducer, mws ...Middleware) Store {
+	return &store{r: r, mws: mws, comps: map[*storeComponent]bool{}}
 }
 
 // Dispatch takes an action and passes it to all middlewares and the Reduce function.
-// It also calls any subscription functions if exist.
-func (s *Store) Dispatch(action interface{}) {
+// NOTE: vecty.Rerender is called in a goroutine for each subscribed component.
+func (s *store) Dispatch(action interface{}) {
```

Comment:
The dispatch logic needs to be in a goroutine to allow blocking calls, this does not effect the state from being updated and rerendered the same as before. I'll update the pull request.